src/words/ast/INodeMovesPredicate.java

```diff
@@ -1,16 +1,57 @@
 package words.ast;
 
+import java.util.HashSet;
+
+import words.environment.WordsAction;
 import words.environment.WordsEnvironment;
+import words.environment.WordsMove;
+import words.environment.WordsObject;
 import words.exceptions.WordsRuntimeException;
 
-public class INodeMovesPredicate extends INode {
+public class INodeMovesPredicate extends INodeBasicActionPredicate {
 	public INodeMovesPredicate(Object... children) {
 		super(children);
 	}
 
 	@Override
 	public ASTValue eval(WordsEnvironment environment) throws WordsRuntimeException {
-		// TODO
-		throw new AssertionError("Not yet implemented");
+		assert false : "Cannot eval INodeMovesPredicate without inherited Statement List";
+		return null;
+	}
+
+	@Override
+	public ASTValue eval(WordsEnvironment environment, Object inheritedStmts) throws WordsRuntimeException {
+		ASTValue subject = children.get(0).eval(environment);
+		ASTValue objectAlias = children.get(1).eval(environment);
+		ASTValue moveDirection = children.get(2) == null ? null : children.get(2).eval(environment);
+		
+		INodeStatementList stmtList = (INodeStatementList) inheritedStmts;
+		
+		HashSet<WordsObject> objectsToCheck = new HashSet<WordsObject>();
+		ASTValue returnVal = new ASTValue(false);
+		
+		// If subject is a string, then we are looking at a class name
+		if (subject.type.equals(ASTValue.ValueType.STRING)) {
```

Comment:
Since you've made INodeBasicActionPredicate an abstract class, this conditional from 34:38 could be separated out as a method that all of the concrete classes use.